src/main/java/com/ibm/cloud/sdk/core/security/IamAuthenticator.java

```diff
@@ -73,7 +75,7 @@ public IamAuthenticator(String apikey) {
    */
   public IamAuthenticator(String apikey, String url, String clientId, String clientSecret,
```

Comment:
I think for Java, we could add a new public ctor that included the new scope parameter since Java allows overloaded functions.